libmamba/src/core/solver.cpp

```diff
@@ -402,6 +402,22 @@ namespace mamba
         return "Encountered problems while solving:\n" + problems.str();
     }
 
+    nlohmann::json MSolver::problems_to_json() const
```

Comment:
Maybe it would be better to have a function `problems(bool all)` that returns a `std::vector<std::string>` with all problems?

Then in another level we can just use that vector of strings to inject it into a nlohmann_json, or do other post-processing (e.g. in Python).

At some point it would also be good to have programmatic access to the underlying data of the problems (just as a FYI). Each problem comes with `type, source, target, dep` where source is the source pacakge, target is the target package an  dep is the dependency connection (as far as I understand).

Reply:
Agree, having a more refined API and corresponding bindings could be pretty useful!

Reply:
Makes sense, all change it!

Reply:
I changed the function to `all_problems` which returns a `std::vector<std::string>`. I imagine that once the error reporting is a bit more fleshed out we can change that to and `std::vector<Problem>` or something like that. I can take a look at that later but this at least unblocks us from adopting the latest version of micromamba.

Reply:
yes, this looks great. For `std::vector<Problem>` -- yep, I think we need to do some additional R&D on how to represent the problems. Libsolv internally uses `Id` to reference everything, but we might want to convert to a type that is easier to access or have some helper functions.